streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -1368,6 +1498,37 @@ class ZStream[-R, +E, +A](val channel: ZChannel[R, Any, Any, Any, E, Chunk[A], A
       .unit
   }
 
+  /**
+   * Like [[ZStream#into]], but provides the result as a [[ZManaged]] to allow for scope
+   * composition.
+   */
+  final def runIntoElementsManaged[R1 <: R, E1 >: E](
+    queue: ZQueue[R1, Nothing, Nothing, Any, Exit[Option[E1], A], Any]
+  ): ZManaged[R1, E1, Unit] = {
+    val writer = ZChannel
+      .managed(self.toPull.mapM(BufferedPull.make[R, E, A](_))) { pull =>
```

Comment:
I think `toPull` is not required here; you could just use `readWith`, no?

Reply:
You're right, done.